src/scripts/main.js

```diff
@@ -1,5 +1,25 @@
 'use strict';
 
 document.addEventListener('click', e => {
-  // write code here
+  const spider = document.querySelector('.spider');
+  const wall = document.querySelector('.wall');
+
+  const spiderRect = spider.getBoundingClientRect();
+  const wallRect = wall.getBoundingClientRect();
+
+  const clickX = e.clientX;
+  const clickY = e.clientY;
+
+  const spiderX = clickX - wallRect.left - spiderRect.width / 2;
```

Comment:
To avoid the spider crossing the wall's edge, adjust the calculation to consider the spider's dimensions. Use Math.min and Math.max to set boundaries.